src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function defaultCompare(a, b) {
+    if (a.toString() < b.toString()) {
+      return -1;
+    }
+
+    if (a.toString() > b.toString()) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    if (this.length === 0) {
+      return this;
+    }
+
+    const compare = compareFunction;
```

Comment:
The variable 'compare' is redundant since 'compareFunction' is already available and not modified. You can directly use 'compareFunction' in the loop.